client/scraper/metadata.py

```diff
@@ -106,26 +108,43 @@ def get_desc(self, response):
                 break
         return desc
 
+    def get_thumbnail(self, response):
+        if self.prop_map[FieldKeyword.IMAGES][FieldKeyword.COUNT] == 0:
+            return None
+        return self.prop_map[FieldKeyword.IMAGES][FieldKeyword.DATA][0]
+
+    def _get_prepend_url(self, image_url):
+        if image_url.startswith('http') or image_url.startswith('www'):
+            return ''
+        return self.prop_map[FieldKeyword.PROVIDER_URL]
+
     def get_images_list(self, response):
         soup = BeautifulSoup(response.content)
         images_list = collections.OrderedDict()
-        image_urls = soup.findAll(MetadataFields.META, attrs={MetadataFields.PROPERTY: MetadataFields.OG_IMAGE})
+        images_list[FieldKeyword.COUNT] = 0
+        images_list[FieldKeyword.DATA] = []
+
+        images = soup.findAll(MetadataFields.META, attrs={MetadataFields.PROPERTY: MetadataFields.OG_IMAGE})
         image_attr = 'content'
-        prepend_url = ''
-        if len(image_urls) == 0:
-            image_urls = soup.findAll('img')
+        if len(images) == 0:
+            images = soup.findAll('img')
             image_attr = 'src'
-            prepend_url = self.prop_map[FieldKeyword.PROVIDER_URL]
-            if len(image_urls) == 0:
+            if len(images) == 0:
                 return None
-        images_list[FieldKeyword.COUNT] = 0
-        images_list[FieldKeyword.DATA] = []
-        for i in range(len(image_urls)):
+
+        for image in images:
             image_item_dict = collections.OrderedDict()
-            if image_urls[i].has_attr(image_attr):
-                image_item_dict[FieldKeyword.URL] = (prepend_url + image_urls[i][image_attr]).encode('utf-8')
-                images_list[FieldKeyword.DATA].append(image_item_dict)
-                images_list[FieldKeyword.COUNT] = images_list[FieldKeyword.COUNT] + 1
+            if image.has_attr(image_attr):
+                image_url = image[image_attr]
+                prepend_url = self._get_prepend_url(image_url)
+                if not (image.has_attr('height') and int(image['height']) < ImageAttrs.MIN_IMAGE_HEIGHT):
+                    if image_url is "":
+                        continue
+                    image_item_dict[FieldKeyword.URL] = (prepend_url + image_url).encode('utf-8')
```

Comment:
there's a url function in `url_utils.py` called `validate_image_url`. Can you call that? it makes sure all the images that are relative urls are fixed.